src/python/tests/test_configuration_manager.py

```diff
@@ -0,0 +1,653 @@
+        result = await get_category_item(category_name='boolean', item_name='info')
+        assert result['description'] == 'boolean type with default False'
+        assert result['type'] == 'boolean'
+        assert result['default'] == 'False'
+        assert result['value'] == 'False'
+
+        await set_category_item_value_entry(category_name='boolean',
+                                            item_name='info', new_value_entry='True')
+        result = await get_category_item(category_name='boolean', item_name='info')
+        assert result['default'] == 'False'
+        assert result['value'] == 'True'
+
+    async def test_get_category_item_empty(self):
+        """
+        Test that gt_category_item when either category_name or item_name do not exist
+        :assert:
+            Assert result is None when category_name or item_name do not exist in configuration
+        """
+        await create_category(category_name='boolean', category_description='boolean type',
+                              category_value={
+                                  'info': {
+                                      'description': 'boolean type with default False',
+                                      'type': 'boolean',
+                                      'default': 'False'}
+                              })
+        result = await get_category_item(category_name='integer', item_name='info')
+        assert result is None
+
+        result = await get_category_item(category_name='boolean', item_name='data')
+        assert result is None
+
+    async def test_get_category_all_items(self):
+        """
+        Test get_category_all_items method returns full "dictionary" of category_value
+        :assert:
+            1.  Values in dictionary are as expected
+            2. default doesn't get updated when value does
+        """
+        await create_category(category_name='boolean', category_description='boolean type',
+                              category_value={
+                                  'info': {
+                                      'description': 'boolean type with default False',
+                                      'type': 'boolean',
+                                      'default': 'False'}
+                              })
+
+        result = await get_category_all_items(category_name='boolean')
+        assert result['info']['description'] == 'boolean type with default False'
+        assert result['info']['type'] == 'boolean'
+        assert result['info']['default'] == 'False'
+        assert result['info']['value'] == 'False'
+
+        await  set_category_item_value_entry(category_name='boolean', item_name='info',
+                                             new_value_entry='True')
+        result = await get_category_all_items(category_name='boolean')
+        assert result['info']['default'] == 'False'
+        assert result['info']['value'] == 'True'
+
+    async def test_get_category_all_items_dne(self):
```

Comment:
does not look a valid test to me! there can be x many things.

Reply:
@MarkRiddoch - There is a question of when doing `get_category_all_items` on a non-existent `category_name`, right now returns `None`; however, an error (to me) makes more sense. 

@praveen-garg - Until Mark reply's I'm commenting this test out.